calico_rkt/calico_rkt.py

```diff
@@ -1,138 +1,149 @@
-
-    # Delete profile
-    try:
-        _datastore_client.remove_profile(input_json['name'])
-    except:
-        print_stderr("Cannot remove profile %s; Profile cannot be found." % container_id)
+                      orchestrator_id=ORCHESTRATOR_ID,
+                      container_id=container_id,
+                      client=_datastore_client)
+
+    profile_name = INPUT_JSON['name']
+
+    # Delete profile if only member
+    if _datastore_client.profile_exists(profile_name) and \
+       len(_datastore_client.get_profile_members(profile_name)) < 1:
+        try:
+            logger.info("Profile %s has no members, removing from datastore" % profile_name)
+            _datastore_client.remove_profile(profile_name)
+        except:
+            logger.info("ERROR: Cannot remove profile %s; Profile cannot be found." % container_id)
+            sys.exit(1)
 
 def _create_calico_endpoint(container_id, netns_path, client):
-    """Configure the Calico interface for a pod."""
-    print_stderr('Configuring Calico networking.', file=sys.stderr)
-
-    interface = env['CNI_IFNAME']
-
-    endpoint, ip = _container_add(hostname=HOSTNAME,
-                                orchestrator_id=ORCHESTRATOR_ID,
-                                container_id=container_id,
-                                netns_path=netns_path,
-                                interface=interface,
-                                client=client)
-    print_stderr('Finished configuring network interface', file=sys.stderr)
-    return endpoint, ip
-
-def _container_add(hostname, orchestrator_id, container_id, netns_path, interface, client):
     """
-    Add a container to Calico networking with the given IP
-    Return Endpoint object
+    Configure the Calico interface for a pod.
+    Return Endpoint and IP
     """
+    logger.info('Configuring Calico networking.')
+
     try:
-        _ = client.get_endpoint(
-            hostname=hostname,
-            orchestrator_id=orchestrator_id,
-            workload_id=container_id
-        )
+        _ = client.get_endpoint(hostname=HOSTNAME,
+                                orchestrator_id=ORCHESTRATOR_ID,
+                                workload_id=container_id)
     except KeyError:
         # Calico doesn't know about this container.  Continue.
         pass
     else:
-        print_stderr("This container has already been configured with Calico Networking.")
+        logger.info("This container has already been configured with Calico Networking.")
         sys.exit(1)
 
+    interface = env['CNI_IFNAME']
+
+    endpoint, ip = _container_add(hostname=HOSTNAME,
+                                  orchestrator_id=ORCHESTRATOR_ID,
+                                  container_id=container_id,
+                                  netns_path=netns_path,
+                                  interface=interface,
+                                  client=client)
+
+    logger.info('Finished configuring network interface')
```

Comment:
might be useful to print the IP and maybe some other information from the endpoint in this logging message